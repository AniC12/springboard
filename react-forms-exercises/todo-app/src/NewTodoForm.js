import React, { useState } from "react";

function NewTodoForm({ addTodo }) {
    const [task, setTask] = useState('');

    const handleChange = (e) => {
        setTask(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo({ task });
        setTask('');
    }

    return (
        <form onSubmit={handleSubmit} data-testid="todo-form">
            <input
                type="text"
                value={task}
                onChange={handleChange}
                placeholder="Enter a new task" />
            <button>Add Todo</button>
        </form>
    );
}

export default NewTodoForm;