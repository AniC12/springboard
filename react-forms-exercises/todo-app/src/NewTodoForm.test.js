import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';
import '@testing-library/jest-dom';

it("renders without crashing", function () {
    render(<NewTodoForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("runs the create function on form submit", function () {
    const createMock = jest.fn();
    render(<NewTodoForm addTodo={createMock} />);
    const createButton = screen.getByText("Add Todo");
    fireEvent.click(createButton);
    expect(createMock).toHaveBeenCalled();
});