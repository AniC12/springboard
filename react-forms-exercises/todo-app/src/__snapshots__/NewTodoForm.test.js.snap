// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    data-testid="todo-form"
  >
    <input
      placeholder="Enter a new task"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
